server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -99,6 +114,20 @@ public class BalancedShardsAllocator implements ShardsAllocator {
         Property.Dynamic,
         Property.NodeScope
     );
+    public static final Setting<Float> INDEXING_TIER_WRITE_LOAD_BALANCE_FACTOR_SETTING = Setting.floatSetting(
+        "cluster.routing.allocation.balance.write_load.indexing",
+        WRITE_LOAD_BALANCE_FACTOR_SETTING,
+        0.0f,
+        Property.Dynamic,
+        Property.NodeScope
+    );
+    public static final Setting<Float> SEARCH_TIER_WRITE_LOAD_BALANCE_FACTOR_SETTING = Setting.floatSetting(
```

Comment:
I thought we'd need a flag (like a feature flag for this new behaviour) and one new setting like the `INDEXING_TIER_SHARD_BALANCE_FACTOR_SETTING` that you have. And that should be enough. I mean we could add `SEARCH_TIER_SHARD_BALANCE_FACTOR_SETTING` to make this cleaner sure. But why do we need `SEARCH_TIER_WRITE_LOAD_BALANCE_FACTOR_SETTING`? That just conceptually doesn't make sense and IMO should be zero if the flag for this behaviour is turned on.

Reply:
Yeah I guess it was just in case we wanted to investigate if write load was somehow a useful predictor of search activity (e.g. if it happens that the most actively updated indices are also the most actively searched, or similar). Perhaps it might be useful to have the option to tweak it to values other than zero?

It seems unlikely, but I didn't want to pre-empt the experiment/investigation. Most likely it'll just be set to zero.